packages/zowe-explorer/src/trees/dataset/DatasetFSProvider.ts

```diff
@@ -218,11 +220,13 @@ export class DatasetFSProvider extends BaseProvider implements vscode.FileSystem
 
         const entryExists = entry != null;
         let entryIsDir = entry != null ? entry.type === vscode.FileType.Directory : false;
+        const uriPath = uri.path.substring(uriInfo.slashAfterProfilePos + 1).split("/");
+        const pdsMember = uriPath.length === 2;
         if (!entryExists) {
-            const resp = await ZoweExplorerApiRegister.getMvsApi(uriInfo.profile).dataSet(path.posix.basename(uri.path), { attributes: true });
+            const resp = await ZoweExplorerApiRegister.getMvsApi(uriInfo.profile).dataSet(uriPath[0], { attributes: true });
             if (resp.success) {
                 const dsorg: string = resp.apiResponse?.items?.[0]?.dsorg;
-                entryIsDir = dsorg?.startsWith("PO") ?? false;
+                entryIsDir = pdsMember ? false : dsorg?.startsWith("PO") ?? false;
```

Comment:
Interestingly enough, the `??` gets evaluated first, then the `? :`
That's why this works without having to add parenthesis.

Curious if we should enforce parenthesis with an eslint rule on mixed operators 😋

Reply:
Hmm.. I intended on the `? :` being evaluated first, but didn't consider that `??` might take priority.
That said, it seems to have the same effect - if `pdsMember` is true, the entry is not a dir - otherwise it will check the dsorg.

Reply:
a quick way to test is:
- prints false: `node  -e "console.log((true ? undefined : null) ?? false)"` 
- prints undefined: `node  -e "console.log(true ? undefined : null ?? false)"`